src/main/java/org/terasology/gooeysQuests/quests/mines/MinesQuestSystem.java

```diff
@@ -0,0 +1,202 @@
+import org.terasology.entitySystem.entity.EntityManager;
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.entitySystem.entity.lifecycleEvents.BeforeDeactivateComponent;
+import org.terasology.entitySystem.event.ReceiveEvent;
+import org.terasology.entitySystem.prefab.Prefab;
+import org.terasology.entitySystem.systems.BaseComponentSystem;
+import org.terasology.entitySystem.systems.RegisterMode;
+import org.terasology.entitySystem.systems.RegisterSystem;
+import org.terasology.gooeysQuests.api.*;
+import org.terasology.logic.inventory.InventoryManager;
+import org.terasology.logic.location.LocationComponent;
+import org.terasology.math.Side;
+import org.terasology.math.geom.Vector3i;
+import org.terasology.registry.In;
+import org.terasology.structureTemplates.events.CheckSpawnConditionEvent;
+import org.terasology.structureTemplates.events.SpawnStructureEvent;
+import org.terasology.structureTemplates.interfaces.BlockPredicateProvider;
+import org.terasology.structureTemplates.interfaces.StructureTemplateProvider;
+import org.terasology.structureTemplates.util.BlockRegionTransform;
+import org.terasology.world.WorldProvider;
+import org.terasology.world.block.Block;
+import org.terasology.world.block.BlockManager;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Random;
+import java.util.function.Predicate;
+
+/**
+ * Makes Gooey offer a Mines quest
+ */
+@RegisterSystem(RegisterMode.AUTHORITY)
+public class MinesQuestSystem extends BaseComponentSystem {
+    private static final int MAX_HORIZONTAL_DISTANCE = 20;
+    private static final int VERTICAL_SCAN_DISTANCE = 20;
+
+    @In
+    private AssetManager assetManager;
+    @In
+    private EntityManager entityManager;
+    @In
+    private WorldProvider worldProvider;
+    @In
+    private BlockManager blockManager;
+    @In
+    private InventoryManager inventoryManager;
+    @In
+    private BlockPredicateProvider blockPredicateProvider;
+    @In
+    private StructureTemplateProvider structureTemplateProvider;
+
+    private Map<EntityRef, MinesQuestSystem.FoundSpawnPossibility> questToFoundSpawnPossibilityMap = new HashMap<>();
+    private Random random = new Random();
+    private Predicate<Block> isAirCondition;
+    private Predicate<Block> isGroundCondition;
+
+    @Override
+    public void postBegin() {
+        isAirCondition = blockPredicateProvider.getBlockPredicate("StructureTemplates:IsAirLike");
+        isGroundCondition = blockPredicateProvider.getBlockPredicate("StructureTemplates:IsGroundLike");
+    }
+
+
+    @ReceiveEvent
+    public void onCreateStartQuestsEvent(CreateStartQuestsEvent event, EntityRef character,
+                                         PersonalQuestsComponent questsComponent) {
+        Prefab questPrefab = assetManager.getAsset("GooeysQuests:MinesQuest", Prefab.class).get();
+        Logger logger = LoggerFactory.getLogger(MinesQuestSystem.class);
+        logger.info("questPrefab: "+questPrefab);
```

Comment:
Trivial whitespace style violations - spaces around the plus here for instance `"+questPrefab`

Also, after you've used logger statements for debugging please remove them unless they are unusually useful and set to the most appropriate log level (this would probably be debug level, not info)